### ISSUE TITLE
Dynamic application of turbo boost [feature proposal]

### ISSUE THREAD

**Issue by garhul** [konkor/cpufreq#217]
#### A bit of bacgkround:
I've been experiencing random shutdowns on my machine (ASUS UM425, ryzen 7 4700u) only in linux, and it seemed to be related to temperatures, unfortunately I wasn't able to make cpufreq work on ubuntu, but after moving to PopOs it worked out of the box, possibly a kernel config issue.

So with PoPos isntalled and cpufreq with disabled frequency boost I experienced no issues, I just fliped the switch on turbo boost and within seconds I saw the temp climb and the machine crash.

I'm assuming that for this type of machine and it's feeble cooling system what windows is doing is allowing for turbo boost to be on for a certain amount of time or within a certain temperature range (as soon as it passes a threshold it gets disabled).

What's the feasibility of implementing such a feature within cpufreq to allow for short bursts of frequency boost within a safe temperature window.

#### TL;DR
Frequency boost on notebooks with poor thermals causes crashes on ryzen cpus, an idea for mitigating such issue would be to implement short bursts of freq boost within safe temperature regions.
